vyper/semantics/analysis/annotation.py

```diff
@@ -231,7 +231,7 @@ def visit_Subscript(self, node, type_):
 
             elif type_ is not None and len(possible_base_types) > 1:
                 for possible_type in possible_base_types:
-                    if isinstance(possible_type.value_type, type(type_)):
+                    if possible_type.value_type == type_:
```

Comment:
a bit pedantic but for hygiene's sake we should use `type_.compare_type(possible_type.value_type)`

also we should panic if no `base_type` is found - you can use the `for.. else` construct for that.